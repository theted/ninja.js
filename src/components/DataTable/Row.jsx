import React from 'react';
import PropTypes from 'prop-types';

const Row = ({ row }) => {
  return (
    <tr>
      <td>
        <a href={row.edit_path}>
          {row.name1}
        </a>
        <br />
        <small>{row.email}</small>
      </td>
    </tr>
  );
};

Row.propTypes = {
  row: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default Row;
