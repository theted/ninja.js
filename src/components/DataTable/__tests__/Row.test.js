import React from 'react';
import { shallow, mount } from 'enzyme';
import Row from '../Row';
import data from '../../../data/users.json';

const row = data[0];

describe('Row component', () => {
  it('renders without crashing', () => {
    shallow(<Row row={row} />);
  });

  it('matches snapshot', () => {
    // TODO: fix table wrap fix
    const component = mount(
      <table>
        <tbody>
          <Row row={row} />
        </tbody>
      </table>,
    );
    expect(component).toMatchSnapshot();
  });
});
