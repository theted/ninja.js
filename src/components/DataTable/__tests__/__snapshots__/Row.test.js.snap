// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component matches snapshot 1`] = `
<table>
  <tbody>
    <Row
      row={
        Object {
          "edit_path": "http://google.com",
          "email": "MadsLKlausen@jourrapide.com",
          "name1": "Mads L. Klausen",
          "per_id": 1,
        }
      }
    >
      <tr>
        <td>
          <a
            href="http://google.com"
          >
            Mads L. Klausen
          </a>
          <br />
          <small>
            MadsLKlausen@jourrapide.com
          </small>
        </td>
      </tr>
    </Row>
  </tbody>
</table>
`;
